phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -2910,6 +3026,16 @@ private static MetaDataMutationResult checkTableKeyInRegion(byte[] key, Region r
                 EnvironmentEdgeManager.currentTimeMillis(), null);
     }
 
+    private static MetaDataMutationResult checkSchemaKeyInRegion(byte[] key, Region region) {
```

Comment:
You should reuse checkTableKeyInRegion and rename it to checkKeyInRegion. You would pass MutationCode to include in the result an extra parameter when the key is not in region. While you are it, also please remove the method checkFunctionKeyInRegion.